### PR TITLE
BigInteger cleanup

### PR DESCRIPTION
**Pull request by hughbe**
Format and cleanup BigInteger - NFC
As I was working on #5974, I saw BigInteger had some quaint formatting
like Complex does.
I went through and fixed some of these small minor changes.
Also, I noticed that ICompare.CompareTo(object) was not covered by any tests. I modified the comparison tests to amend this.

### CODE REVIEW

**Comment by hughbe** [comment]
Tests are failing for unrelated reasons - CI bug?

**Comment by mellinoe** [comment]
> Tests are failing for unrelated reasons - CI bug?

I think we're having disk space issues in our CI system right now. We can restart the ones that have issues when everything looks clean

File: src/Common/src/System/Globalization/FormatProvider.Number.cs [hunk 1]
```diff
@@ -708,10 +705,9 @@ internal static unsafe string NumberToString(NumberBuffer number, char format, i
 
                 StringBuilder sb = new StringBuilder(MIN_SB_BUFFER_SIZE);
 
-                switch (format)
+                switch (char.ToUpper(format))
```
Comment (stephentoub) [inline src/Common/src/System/Globalization/FormatProvider.Number.cs]: Please revert this change.  The other changes thus far have just been about style, which is fine.  This one is a functional change, and while it's very likely the same performance, there is no validation for that, and the value of the change is low.

File: src/System.Runtime.Numerics/src/System/Numerics/BigIntegerCalculator.AddSub.cs [hunk 2]
```diff
@@ -58,9 +58,7 @@ public unsafe static uint[] Add(uint[] left, uint[] right)
         }
 
         [SecuritySafeCritical]
-        private unsafe static void Add(uint* left, int leftLength,
-                                       uint* right, int rightLength,
-                                       uint* bits, int bitsLength)
+        private unsafe static void Add(uint* left, int leftLength, uint* right, int rightLength, uint* bits, int bitsLength)
```
Comment (stephentoub) [inline src/System.Runtime.Numerics/src/System/Numerics/BigIntegerCalculator.AddSub.cs]: Please revert these changes.  The previous formatting helped to make the meaning of the parameter more clear, with each line containing the pointer and the associated length.  Please revert all similar changes that were made.

File: src/System.Runtime.Numerics/src/System/Numerics/BigIntegerCalculator.BitsBuffer.cs [hunk 3]
```diff
@@ -8,9 +8,8 @@
 namespace System.Numerics
 {
     // ATTENTION: always pass BitsBuffer by reference,
-    // it's a structure for performance reasons. Furthermore
+    // it's a class for performance reasons. Furthermore
```
Comment (stephentoub) [inline src/System.Runtime.Numerics/src/System/Numerics/BigIntegerCalculator.BitsBuffer.cs]: This whole comment can be deleted.  It's no longer relevant.

File: src/System.Runtime.Numerics/src/System/Numerics/BigIntegerCalculator.PowMod.cs [hunk 4]
```diff
@@ -3,17 +3,16 @@
...
 using System.Diagnostics;
-using System.Security;
 
 namespace System.Numerics
 {
+    /// <summary>
+    // Executes different exponentiation algorithms, which are
+    // basically based on the classic square-and-multiply method.
+    // https://en.wikipedia.org/wiki/Exponentiation_by_squaring
+    /// </summary>
```
Comment (stephentoub) [inline src/System.Runtime.Numerics/src/System/Numerics/BigIntegerCalculator.PowMod.cs]: This comment was about the Pow method, not about the BigIntegerCalculator class.

**Comment by stephentoub** [comment]
Thanks, @hughbe.  Mostly looks fine, but some things that need to be reverted before this can be merged.

**Comment by hughbe** [comment]
@stephentoub  thanks for the comments. I followed up on all of them, even though I do disagree with the formatting reverts, but thats not really very important!

Enjoy!

File: src/System.Runtime.Numerics/src/System/Numerics/BigIntegerCalculator.SquMul.cs [hunk 5]
```diff
@@ -359,9 +359,7 @@ public unsafe static uint[] Multiply(uint[] left, uint[] right)
         }
 
         [SecuritySafeCritical]
-        private unsafe static void SubtractCore(uint* left, int leftLength,
-                                                uint* right, int rightLength,
-                                                uint* core, int coreLength)
+        private unsafe static void SubtractCore(uint* left, int leftLength, uint* right, int rightLength, uint* core, int coreLength)
```
Comment (stephentoub) [inline src/System.Runtime.Numerics/src/System/Numerics/BigIntegerCalculator.SquMul.cs]: This one should be reverted, too.

File: src/System.Runtime.Numerics/src/System/Numerics/BigIntegerCalculator.PowMod.cs [hunk 6]
```diff
@@ -3,17 +3,16 @@
...
 {
     internal static partial class BigIntegerCalculator
     {
+        /// <summary>
         // Executes different exponentiation algorithms, which are
         // basically based on the classic square-and-multiply method.
-
         // https://en.wikipedia.org/wiki/Exponentiation_by_squaring
-
+        /// </summary>
```
Comment (stephentoub) [inline src/System.Runtime.Numerics/src/System/Numerics/BigIntegerCalculator.PowMod.cs]: All of the lines of the XML comment need three slashes rather than two.

**Comment by hughbe** [comment]
@stephentoub responded to your PR feedback for these last 2 comments

**Comment by stephentoub** [comment]
@dotnet-bot test this please

**Comment by stephentoub** [comment]
@dotnet-bot test this please

File: src/Common/src/System/Globalization/FormatProvider.Number.cs [hunk 7]
```diff
@@ -1364,14 +1359,14 @@ internal static unsafe string NumberToStringFormat(NumberBuffer number, string f
...
                         int groupTotalSizeCount = 0;
-                        int groupSizeLen = groupDigits.Length;    // the length of groupDigits array.
+                        int groupSizeLen = groupDigits.Length;    // The length of groupDigits array.
                         if (groupSizeLen != 0)
-                            groupTotalSizeCount = groupDigits[groupSizeIndex];   // the current running total of group size.
+                            groupTotalSizeCount = groupDigits[groupSizeIndex];   // The current running total of group size.
                         int groupSize = groupTotalSizeCount;
 
-                        int totalDigits = digPos + ((adjust < 0) ? adjust : 0); // actual number of digits in o/p
+                        int totalDigits = digPos + ((adjust < 0) ? adjust : 0); // Actual number of digits in o/p
```
Comment (stephentoub) [inline src/Common/src/System/Globalization/FormatProvider.Number.cs]: FYI, for future cleanup, there's no need to change comments like this to start with a capital... it's fine having short descriptions that aren't full sentences.

**Comment by stephentoub** [comment]
@hughbe, in the future please refrain from moving a whole bunch of code around.  It leads to lots of changes that make it difficult to see in a diff and doesn't provide a lot of value.  If such changes are going to be made, they should be done in independent commits where _nothing_ else changes.  The diff was so large as a result of these moves that I initially missed the fact that GitHub wasn't even showing a diff for BitInteger.cs; I had to review it offline in Visual Studio.

**Comment by stephentoub** [comment]
BigInteger.cs L1173: Why did you change the implementations of the ToString methods?  Note that you're now passing string.Empty where null used to be passed.  It likely doesn't make a difference, but I don't see the value in these changes.

**Comment by stephentoub** [comment]
BigInteger.cs L1715: You moved where ```bits``` is defined so that it's now at a larger scope.  This is more error prone and more likely to accidentally have errors.  Please revert this so that the variable is defined where it's used, even though that means multiple blocks will have it defined.

**Comment by stephentoub** [comment]
BigInteger.cs L1764.  Same as above with the declaration of ```bits```.

**Comment by stephentoub** [comment]
Thanks, @hughbe.  A few more changes to be addressed.  You can do them in the last "PR feedback" commit.  Thanks.

**Comment by hughbe** [comment]
@stephentoub 
Sorry, I was just following StyleCop guidelines :)

Thanks for these comments, I've rectified 'em.

**Comment by stephentoub** [comment]
Thanks, @hughbe. LGTM.
